script/upload

```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
+
+set -e
+
+REPO="git-lfs/git-lfs"
+WORKDIR="$(mktemp -d)"
+
+trap 'rm -fr "$WORKDIR"' EXIT
+
+say () {
+  [ -n "$QUIET" ] && return
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+}
+
+abort () {
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+  exit 2
+}
+
+curl () {
+  command curl -nfSs "$@"
+}
+
+categorize_os () {
+  local os="$1"
+
+  if [ "$os" = "freebsd" ]
+  then
+    echo FreeBSD
+  else
+    ruby -e 'puts ARGV[0].capitalize' "$os"
+  fi
+}
+
+categorize_arch () {
+  local arch="$1"
+
+  ruby -e 'puts ARGV[0].upcase' "$arch"
+}
+
+# Categorize a release asset and print its human readable name to standard
+# output.
+categorize_asset () {
+  local file="$1"
+  local os=$(echo "$file" | sed -e 's/^git-lfs-//' -e 's/[-.].*$//')
+  local arch=$(echo "$file" | ruby -pe '$_.gsub!(/\Agit-lfs-[^-]+-([^-]+)[-.].*/, "\\1")')
+
+  if ruby -e 'exit 1 unless /\Agit-lfs-v\d+\.\d+\.\d+\.tar\.gz\z/ =~ ARGV[0]' "$file"
+  then
+    echo "Source"
+  elif ruby -e 'exit 1 unless /\Agit-lfs-windows-v\d+\.\d+\.\d+\.exe\z/ =~ ARGV[0]' "$file"
+  then
+    echo "Windows Installer"
+  elif [ "$file" = "sha256sums.asc" ]
+  then
+    echo "Signed SHA-256 Hashes"
+  else
+    printf "%s %s\n" "$(categorize_os "$os")" "$(categorize_arch "$arch")"
+  fi
+}
+
+# Provide a content type for the asset based on its file name.
+content_type () {
+  local file="$1"
+
+  if [ -z "${file%%*.zip}" ]
+  then
+    echo "application/zip"
+  elif [ -z "${file%%*.tar.gz}" ]
+  then
+    echo "application/gzip"
+  elif [ -z "${file%%*.exe}" ]
+  then
+    echo "application/octet-stream"
+  else
+    echo "text/plain"
+  fi
+}
+
+# Format the JSON for creating the release and print it to standard output.
+format_release_json () {
+  local version="$1"
+  local bodyfile="$2"
+
+  ruby -rjson -e 'puts JSON.generate({
```

Comment:
Hmmm. This one is tricky, too: again I think that it would be beneficial to not have a dependency on Ruby here, but we might be able to get out easily since we _do_ assume that `jq` is installed. I'm not sure of how to do this either, but it seems like it ought to be possible.